Add build options for ROOT 6.38 and improve NixOS install instructions

guitargeek opened the pull request.

guitargeek committed:
Don't mention `nix-env` in ROOT installation instructions

The `nix-env` command is deprecated because it creates non-reproducible
environments. Also, explain how to use the Python package.

vepadulano commented:
The build options table should be generated via the script during the release process

guitargeek committed:
Raise minimum required CMake version to 3.20

This mimics the main repo, and avoids annoying errors if people
copy-paste the code and run it with newer CMake versions that dropped
compatibility with CMake < 3.5.
